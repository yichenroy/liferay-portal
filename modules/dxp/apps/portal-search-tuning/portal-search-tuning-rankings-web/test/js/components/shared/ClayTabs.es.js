/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * The contents of this file are subject to the terms of the Liferay Enterprise
 * Subscription License ("License"). You may not use this file except in
 * compliance with the License. You can obtain a copy of the License by
 * contacting Liferay, Inc. See the License for the specific language governing
 * permissions and limitations under the License, including but not limited to
 * distribution rights of the Software.
 */

import {
	ClayTab,
	ClayTabList,
	ClayTabPanel,
	ClayTabs
} from '../../../../src/main/resources/META-INF/resources/js/components/shared/ClayTabs.es';
import React from 'react';
import {render} from '@testing-library/react';

describe('ClayTabs', () => {
	it('renders', () => {
		const {asFragment} = render(
			<ClayTabs>
				<ClayTabList className="results-ranking-tabs">
					<ClayTab>Tab 1</ClayTab>

					<ClayTab>Tab 2</ClayTab>
				</ClayTabList>

				<ClayTabPanel>Tab Panel 1</ClayTabPanel>
				<ClayTabPanel>Tab Panel 2</ClayTabPanel>
			</ClayTabs>
		);

		expect(asFragment()).toMatchSnapshot();
	});
});
