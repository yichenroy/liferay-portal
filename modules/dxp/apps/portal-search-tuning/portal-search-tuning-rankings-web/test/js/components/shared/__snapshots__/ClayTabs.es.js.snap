// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClayTabs renders 1`] = `
<DocumentFragment>
  <div
    class="react-tabs"
    data-tabs="true"
  >
    <ul
      class="nav nav-underline results-ranking-tabs"
      role="tablist"
    >
      <li
        aria-controls="react-tabs-1"
        aria-disabled="false"
        aria-selected="true"
        class="nav-item react-tabs__tab--selected"
        id="react-tabs-0"
        role="tab"
        tabindex="0"
      >
        <button
          class="btn btn-unstyled nav-link active"
        >
          Tab 1
        </button>
      </li>
      <li
        aria-controls="react-tabs-3"
        aria-disabled="false"
        aria-selected="false"
        class="nav-item"
        id="react-tabs-2"
        role="tab"
      >
        <button
          class="btn btn-unstyled nav-link"
        >
          Tab 2
        </button>
      </li>
    </ul>
    <div
      aria-labelledby="react-tabs-0"
      class="react-tabs__tab-panel react-tabs__tab-panel--selected"
      id="react-tabs-1"
      role="tabpanel"
    >
      Tab Panel 1
    </div>
    <div
      aria-labelledby="react-tabs-2"
      class="react-tabs__tab-panel"
      id="react-tabs-3"
      role="tabpanel"
    />
  </div>
</DocumentFragment>
`;
